src/test/ui/consts/trait_specialization.rs

```diff
@@ -0,0 +1,64 @@
+// ignore-wasm32-bare which doesn't support `std::process:exit()`
+// run-pass
+
+// Tests that specialization does not cause optimizations running on polymorphic MIR to resolve
+// to a `default` implementation.
+
+#![feature(specialization)]
+
+trait Marker {}
+
+trait SpecializedTrait {
+    const CONST_BOOL: bool;
+    const CONST_STR: &'static str;
+    fn method() -> &'static str;
+}
+impl <T> SpecializedTrait for T {
```

Comment:
Doesn't rustfmt trigger on this?

```suggestion
impl<T> SpecializedTrait for T {
```

Reply:
Just ran the formatter